[anshumaantgi] iP

anshumaantgi opened the pull request:
# Ava
>  Why forget , when Ava can keep track of it ? - The Creator of Ava

Ava frees your mind of having to remember things you need to do. It's
- text-based
- easy to learn 
- ~~FAST~~ SUPER FAST to use 

All you need to do is ,
1. Download it from [here](https://github.com/anshumaantgi/ip)
2. double-click it.
3. add you tasks.
4. let it manage your task for 😉 

And it's **FREE**

Features:
- [x] Managing Task
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

if you're a Java programmer, you can. use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

anshumaantgi committed:
Add AvaUI (Ava's user interface) , rename Duke to Ava

anshumaantgi committed:
Add Task,TaskList
Move Intialisation to AvaUI

anshumaantgi committed:
Modified Task, TaskList to accomadate mark/unmark feature

anshumaantgi committed:
Add Deadline, Event and Todo
Modify TaskList to identify type of event

anshumaantgi committed:
Added Tests

anshumaantgi committed:
Add Exceptions

anshumaantgi committed:
Add delete functionality
Refactor Code

anshumaantgi committed:
Add Storage.java , Modify Task and it's subclasses to add storage representation , Modify TaskList to save changes

anshumaantgi committed:
Add TimeParser, modify Event,Deadline to contain timeParser Object

anshumaantgi committed:
Add TimeParser , Remove unwated try-catch from Ava-UI

anshumaantgi committed:
Merge branch 'branch-Level-7' into branch-Level-8

# Conflicts:
#	src/main/java/AvaUI.java
#	src/main/java/TaskList.java

anshumaantgi committed:
Merge branch 'branch-Level-8'

anshumaantgi committed:
Add Commands(AddTask,MarkTask,UnmarkTask,Exit,DeleteTask,ListTask) ,Add Parser
Modify AvaUi -> Ui (Make the object methods static)

anshumaantgi committed:
Merge pull request #1 from anshumaantgi/add-gradle-support

Add Gradle support

anshumaantgi committed:
Add Gradle Support

anshumaantgi committed:
Add Test Files

anshumaantgi committed:
Add JavaDoc comments

anshumaantgi committed:
Modify According to Coding Standard

anshumaantgi committed:
Add Coding Standard

anshumaantgi committed:
Add Find Functionality

new Functionality.

Added New Command (: FindTask.java).

new functionality requires new command.

anshumaantgi committed:
Merge branch 'branch-A-CodingStandard'

anshumaantgi committed:
Resolve Merge Conflicts between A-CodingStandard and Master

anshumaantgi committed:
Merge branch 'branch-Level-9'

anshumaantgi committed:
Resolve conflicts between branches Level-9 and Master

Tempura-Person commented on src/main/java/Ava/Commands/Exit.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Override
+    public String output(String formatSpace){
+        return "Will Never reach here";
```
Should probably be a comment instead of a String return?

Tempura-Person commented on src/main/java/Ava/Commands/Exit.java:
```diff
@@ -0,0 +1,27 @@
+     */
+    @Override
+    public boolean run(TaskList t, Storage s){
+        return false;
```
If all the run method does is return false, and no special 'goodbye' message is printed here, is this class necessary?

Tempura-Person commented on src/main/java/Ava/Commands/FindTask.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     *
```
Seems to be an extra newline here in the javadocs.

Tempura-Person commented on src/main/java/Ava/Exceptions/AvaException.java:
```diff
@@ -0,0 +1,6 @@
+package Ava.Exceptions;
```
I like how you made several Exception classes to handle different exception cases.
However, they do seem a little all over the place in terms of the naming convention.
(e.g. CommandNotFoundException should probably be named CommandNotFound, to fit with the other ones.)

Tempura-Person commented on src/main/java/Ava/Exceptions/CommandNotFoundException.java:
```diff
@@ -0,0 +1,14 @@
+package Ava.Exceptions;
+
+public class CommandNotFoundException extends AvaException {
+    private final String ERROR_PROMPT = "Command Not Complete or Possible !!";
```
The other Exception classes seem to give a clear indicator of what the correct way is for the user to input text.
Perhaps this Exception could tell the user the correct command formats?

Tempura-Person reviewed (commented):
Quite a solid IP, there was a good amount of thought put into structuring the individual Classes.
However, I am not sure why there is a duplicate directory for Ava.
(there is both an Ava file and an ava file, containing Class duplicates.)

anshumaantgi committed:
Delete src/main/java/Ava directory

Delete Extra Ava

anshumaantgi committed:
Delete src/test/java/Ava/Commands directory

Delete Extra Test Commands

anshumaantgi committed:
Fix Git Issues

anshumaantgi committed:
Fix Git Issues

Modifying Files so concurrent Merge does not delete them

anshumaantgi committed:
Merge commit '7d4000e2cc12512267fc1550a795d934f736f6d5'

# Conflicts:
#	src/main/java/Ava/Ava.java
#	src/main/java/Ava/Commands/AddTask.java
#	src/main/java/Ava/Commands/AvaCommand.java
#	src/main/java/Ava/Commands/DeleteTask.java
#	src/main/java/Ava/Commands/Exit.java
#	src/main/java/Ava/Commands/FindTask.java
#	src/main/java/Ava/Commands/ListTask.java
#	src/main/java/Ava/Commands/MarkTask.java
#	src/main/java/Ava/Commands/UnmarkTask.java
#	src/main/java/Ava/Exceptions/AvaException.java
#	src/main/java/Ava/Exceptions/CommandNotFoundException.java
#	src/main/java/Ava/Exceptions/DateTimeNotParsed.java
#	src/main/java/Ava/Exceptions/NonExistentTask.java
#	src/main/java/Ava/Parser.java
#	src/main/java/Ava/Storage.java
#	src/main/java/Ava/TaskList.java
#	src/main/java/Ava/Tasks/Deadline.java
#	src/main/java/Ava/Tasks/Event.java
#	src/main/java/Ava/Tasks/Task.java
#	src/main/java/Ava/Tasks/TimeParser.java
#	src/main/java/Ava/Tasks/Todo.java
#	src/main/java/Ava/Ui.java
#	src/test/java/Ava/Commands/AddTaskTest.java
#	src/test/java/Ava/Commands/MarkTaskTest.java

anshumaantgi committed:
Fix Github Issues

anshumaantgi committed:
Delete src/main/java/Ava directory

Merge into 1 Folder

anshumaantgi committed:
Delete src/main/java/ava directory

Merge into 1 Folder

anshumaantgi committed:
Rename to Merge Two Folders

anshumaantgi commented:
> Quite a solid IP, there was a good amount of thought put into structuring the individual Classes. However, I am not sure why there is a duplicate directory for Ava. (there is both an Ava file and an ava file, containing Class duplicates.)

> Thanks for the feedback , I have removed the two folders and just left the one .

anshumaantgi committed:
Add CheckStyle Files

-> Inside the Config Directories

anshumaantgi committed:
Add GUI to the Project

anshumaantgi committed:
Ava Improving GUI

anshumaantgi committed:
Add assertions

No way of checking If the commands checking correct input function runs or not .
Later on if the code is changed , invalid argument could be sent to the run function of the commands.
Let's add assertions to check if valid arguments are being passed i.e. checking this.correctInput() function runs.
Assertion terminate indicating invalid argument.

anshumaantgi committed:
Improve CodeQuality

Code had multiple violation to the required coding standard and on multiple occasions exceptions were on different abstraction levels
Exception were aligned to be on single abstraction level in commands
Wildcard imports were changed to specify the classes being imported.
Done this way to improve code quality.

anshumaantgi committed:
Merge pull request #3 from anshumaantgi/branch-A-Assertions

Add assertions
No way of checking If the commands checking correct input function runs or not .
Later on if the code is changed , invalid argument could be sent to the run function of the commands.
Let's add assertions to check if valid arguments are being passed i.e. checking this.correctInput() function runs.
Assertion terminate indicating invalid argument.

anshumaantgi committed:
Fix CodeQuality

running was used a boolean , thus the verb needed to be changed.
running variable name was changed

anshumaantgi committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Ava/Parser.java
#	src/main/java/Ava/commands/AddTask.java
#	src/main/java/Ava/commands/DeleteTask.java
#	src/main/java/Ava/commands/FindTask.java
#	src/main/java/Ava/commands/MarkTask.java
#	src/main/java/Ava/commands/UnmarkTask.java

anshumaantgi committed:
Merge pull request #4 from anshumaantgi/branch-A-CodeQuality

Improve CodeQuality

anshumaantgi committed:
Implement C-Sort

Allow User to Sort current list
Organise Task as before Tasks were unorganised

anshumaantgi committed:
Add UserGuide and Ui.png

anshumaantgi committed:
Fix Jar File Error

anshumaantgi committed:
Update README.md

Make ReadMe.md more readable
